### ISSUE TITLE
IBVS PointsBased not working in Gazebo

### ISSUE THREAD

**Issue by aarontan-git** [PamirGhimire/visualServoing_ROSProject#3]
Hello,

I am finally able to make and run the IBVS PointsBased package. 

However, after selecting the blobs in a counter clock wise fashion starting with the lower left blob, the turtlebot does not seem to move closer to the target. 

Its reactions thus far has been sudden jerk movement and sometimes slowly moving backwards.

I am wondering, did you guys solve this issue with your IBVS controller? Is this a problem with the gain value?

Please help.

PS. 

In the readme file for IBVSpointsbased, you asked the users to enter the following to run the sim:
rosrun IBVS_PointsBased IBVS_PointsBased_gazebo

however,  to run the simulation in gazebo, you need to run the following instead:
rosrun project_vs IBVS_PointsBased_gazebo

**Comment by PamirGhimire** [comment]
We did see jerky motion with the real robot, never in the simulation, this
remained unresolved

**Comment by PamirGhimire** [comment]
This  was less frequent than the controller actually working

**Comment by aarontan-git** [comment]
This is the problem that I am facing, have you seen something like this before?

https://youtu.be/tYc_KH50I_Y

**Comment by PamirGhimire** [comment]
right, i remember i had to check the polarity for each velocity dof
separately, they were different for the real and virtual bots,i suggest you
do that as well,` also, when running this on the real bot, make sure that
you're subscribed to mono_image, not rgb as in gazebo

**Comment by Abduoit** [comment]
@PamirGhimire 
what do u mean by checking the polarity, how to do this ?

there is no image_mono topic in gazebo, there is only topic `/camera/rgb/image_raw`

**Comment by PamirGhimire** [comment]
when working with the real robot, you might find image_mono as well, by
'checking polarity', for example for rotation around the z-axis, make all
but v[z-axis-rot-vel-indx] zeros, try +/- rotation-vel values for this dof,
see which one produces the expected behavior (for example: v[5] = 0.3 or
-0.3 makes it rotate clockwise, same for front-back)

**Comment by AnukritiSinghh** [comment]
@Abduoit hey, can you share your exact commands used to run this? I am running into an error while catkin_make of pointbased vip and I am not able to solve it.
Thanks

![image](https://user-images.githubusercontent.com/28527591/146422472-7af246bb-a655-409c-b4d5-14e799c6de18.png)
